src/setfit/trainer.py

```diff
@@ -69,7 +70,7 @@ def __init__(
         train_dataset: "Dataset" = None,
         eval_dataset: "Dataset" = None,
         model_init: Callable[[], "SetFitModel"] = None,
-        metric: str = "accuracy",
+        metric: Union[str, Callable] = "accuracy",
```

Comment:
Would it be possible to be more specific with the types that Callable accepts?

see https://docs.python.org/3/library/typing.html#callable

Reply:
Good point, sure